bgpd/bgp_updgrp_packet.c

```diff
@@ -523,11 +523,16 @@ struct stream *bpacket_reformat_for_peer(struct bpacket *pkt,
 			gnh_modified = 1;
 		}
 
-		if (IN6_IS_ADDR_UNSPECIFIED(mod_v6nhg)) {
-			if (peer->nexthop.v4.s_addr != INADDR_ANY) {
-				ipv4_to_ipv4_mapped_ipv6(mod_v6nhg,
-							 peer->nexthop.v4);
-			}
+		if (peer->nexthop.v4.s_addr != INADDR_ANY &&
+		    (IN6_IS_ADDR_UNSPECIFIED(mod_v6nhg) ||
+		     (peer->connection->su.sa.sa_family == AF_INET &&
```

Comment:
Shouldn't it only be `(IN6_IS_ADDR_UNSPECIFIED(mod_v6nhg) && (peer->connection->su.sa.sa_family == AF_INET && paf->afi == AFI_IP6)))`

Reply:
No it is correct.

When it is an IPv6 prefix over an IPv4 BGP session and the outgoing interface to the peer has a standard IPv6 address, `mod_v6nhg` contains this standard IPv6 address.

`IN6_IS_ADDR_UNSPECIFIED(mod_v6nhg)` is for the case of an IPv4 prefix over an IPv6 unnumbered BGP session. See `tests/topotests/bgp_ipv4_over_ipv6/test_rfc5549_ebgp_unnumbered_nbr.py` -> 'def bgp_prefix_received_v4_mapped_v6_nh(router)'